### PR TITLE
{Pylint} Fix arguments-renamed

### PR DESCRIPTION
**Pull request by jiasli**
**Description**
Finish the unfinished tasks of https://github.com/Azure/azure-cli/pull/20192

Fix [`arguments-renamed`](https://pylint.readthedocs.io/en/stable/user_guide/messages/warning/arguments-renamed.html) pylint error.

### CODE REVIEW

**Comment by yonzhan** [comment]
Thank you for your contribution! We will review the pull request and get back to you soon.

File: src/azure-cli/azure/cli/command_modules/ams/_utils.py [hunk 1]
```diff
@@ -60,22 +60,22 @@ class JsonBytearrayEncoder(json.JSONEncoder):
     DATE_FORMAT = "%Y-%m-%d"
     TIME_FORMAT = "%H:%M:%S"
 
-    def default(self, obj):  # pylint: disable=E0202,W0221
```
Comment (jiasli) [inline src/azure-cli/azure/cli/command_modules/ams/_utils.py:63]: `W0221` is `arguments-differ`: https://pylint.readthedocs.io/en/latest/user_guide/messages/warning/arguments-differ.html

File: src/azure-cli-core/azure/cli/core/commands/parameters.py [hunk 2]
```diff
@@ -365,8 +365,7 @@ def _ignore_if_not_registered(self, dest):
         if not match:
             super().argument(dest, arg_type=ignore_type)
 
-    # pylint: disable=arguments-differ
```
Comment (jiasli) [inline src/azure-cli-core/azure/cli/core/commands/parameters.py:368]: Putting `# pylint: disable=arguments-differ` here is incorrect as it disables all `arguments-differ` below, not only for `argument()`.

File: src/azure-cli-core/azure/cli/core/commands/parameters.py [hunk 3]
```diff
@@ -467,15 +466,17 @@ def _expansion_validator_impl(namespace):
                       options_list=dest_option,
                       validator=get_complex_argument_processor(expanded_arguments, dest, model_type))
 
-    def ignore(self, *args):
+    def ignore(self, *args):  # pylint: disable=arguments-differ
+        # It is expected that this method's signature differs from its base class, as it can be used to ignore
+        # multiple arguments in one method call.
```
Comment (jiasli) [inline src/azure-cli-core/azure/cli/core/commands/parameters.py:471]: Some examples:

https://github.com/Azure/azure-cli/blob/464a79cb0f9c474da1f9d426b9aaf56afcffc47a/src/azure-cli/azure/cli/command_modules/vm/_params.py#L272

https://github.com/Azure/azure-cli/blob/0b06b4f295766bcadaebdb7cf8fc05c7d6c9a5a8/doc/authoring_command_modules/authoring_commands.md#L292

**Review by evelyn-ys — APPROVED** [review]

**Review by evelyn-ys — APPROVED** [review]

**Pull request merged**